text/0029-hammer-help-formatting.md

```diff
@@ -0,0 +1,260 @@
+See following chunk from `hammer repository list -h`:
+```
+#...
+--content-view CONTENT_VIEW_NAME                                            Content view name to search by
+--content-view-id CONTENT_VIEW_ID                                           content view numeric identifier
+--content-view-version CONTENT_VIEW_VERSION_VERSION                         Content view version number
+--content-view-version-content-view-id CONTENT_VIEW_VERSION_CONTENT_VIEW_ID Content view to search by
+--content-view-version-id CONTENT_VIEW_VERSION_ID                           Content view version identifier
+--environment ENVIRONMENT_NAME                                              Name to search by
+--environment-id ENVIRONMENT_ID
+#...
+--organization ORGANIZATION_NAME                                            Organization name to search by
+--organization-id ORGANIZATION_ID                                           organization ID
+--organization-label ORGANIZATION_LABEL                                     Organization label to search by
+#...
+```
+
+Also some parameter descriptions (either generated or from apidocs) don't really add more information and only repeat
+the option name:
+```
+--organization ORGANIZATION_NAME          Organization name
+--organization-id ORGANIZATION_ID         organization ID
+```
+Such options are quite self-explanatory and the only benefit for keeping the descriptions is that they're translated.
+It might make sense to remove such descriptions.
+
+
+### 4. non-uniform capitalization in the descriptions
+
+We have different capitalization in option descriptions accross the whole hammer. See following part of `hammer repository create -h`:
+```
+#...
+--upstream-password UPSTREAM_PASSWORD       Password of the upstream repository user used for authentication
+--upstream-username UPSTREAM_USERNAME       Username of the upstream repository user used for authentication
+--url URL                                   repository source url
+--verify-ssl-on-sync VERIFY_SSL_ON_SYNC     if true, Katello will verify the upstream url’s SSL certifcates are signed by a trusted CA.
+                                            One of true/false, yes/no, 1/0.
+-h, --help                                  print help
+```
+
+# Proposed solutions
+[solution]: #proposed-solutions
+
+### 1. long option lists are difficult to read
+
+Resource identifier options can be squeezed to one line.
+
+```
+before:
+--architecture ARCHITECTURE_NAME       Architecture name
+--architecture-id ARCHITECTURE_ID
+
+after:
+--architecture[-id] ARCHITECTURE       Architecture
+```
+
+```
+before:
+--organization ORGANIZATION_NAME         Organization name to search by
+--organization-id ORGANIZATION_ID        organization ID
+--organization-label ORGANIZATION_LABEL  Organization label to search by
+
+after:
+ --organization[-id|-label] ORGANIZATION  Organization
+```
+
```

Comment:
Interesting idea. It would require more changes than just help but it's definitely doable and could be easier to use.

Two drawbacks I can see:

- It creates more requests.
- If you create a record called `"12"` (not that it would be wise) and there's already one with `id = 12` you're lost. But that might be a corner case we can live with.